### PR TITLE
ASoC: SOF: imx: add missing MODULE_LICENSE() for imx-common

### PR DESCRIPTION
**Pull request by kv2019i**
Fix build warning:
WARNING: modpost: missing MODULE_LICENSE() in sound/soc/sof/imx/imx-common.o

Fixes: 18ebffe4d043 ("ASoC: SOF: imx: Add debug support for imx platforms")
Reported-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

### CODE REVIEW

**Review by dbaluta — DISMISSED** [review]

**Comment by dbaluta** [comment]
thanks @kv2019i

**Review by paulstelian97 — DISMISSED** [review]
Nice, thanks. Guess we missed it because we don't use any GPL symbols?

**Comment by kv2019i** [comment]
@paulstelian97 wrote:
> Nice, thanks. Guess we missed it because we don't use any GPL symbols?

Hmm, I think you get this warning on every build. I think we have a bug in our kconfig magic as the IMX build does not get enabled at all (!). E.g. Travis build of "bash kconfig/kconfig-sof-default.sh" in in https://travis-ci.org/github/thesofproject/linux/jobs/728133077 -- I don't see SOF IMX modules built at all, although they should be.

**Comment by paulstelian97** [comment]
Will take a quick look later today to see what needs to be done. Thanks for pointing this out.

**Comment by kv2019i** [comment]
@kv2019i wrote:
 > Hmm, I think you get this warning on every build. I think we have a bug in our kconfig magic as the IMX build does not get enabled at all (!). E.g. Travis build of "bash kconfig/kconfig-sof-default.sh" in in https://travis-ci.org/github/thesofproject/linux/jobs/728133077 -- I don't see SOF IMX modules built at all, although they should be.

Hmm, my bad. IMX is only built on ARM64 and in our ARM64 builds IMX is enabled. The warning is there, but Travis does not pick it up:
https://travis-ci.org/github/thesofproject/linux/jobs/728133080
```
WARNING: modpost: missing MODULE_LICENSE() in sound/soc/sof/imx/imx-common.o
```

**Comment by kv2019i** [comment]
And my patch actually fails on AMD64 now in Travis:
https://travis-ci.org/github/thesofproject/linux/jobs/728274035

Sending a new attempt in a minute.

**Comment by paulstelian97** [comment]
Yeah iMX only has ARM64 for now, no Intel platforms supported now or to my knowledge planned to be supported in the future.

**Comment by kv2019i** [comment]
Ok, now the build is clean. amd64 build is green and no warning about modpost:
https://travis-ci.org/github/thesofproject/linux/jobs/728315675

**Comment by paulstelian97** [comment]
Warning from Checkpatch (does CI do shallow pull and in reality that commit exists just fine?)

**Review by paulstelian97 — APPROVED** [review]
Approve conditional on that Checkpatch issue being a faux issue.

**Comment by kv2019i** [comment]
@paulstelian97 wrote:
> Warning from Checkpatch (does CI do shallow pull and in reality that commit exists just fine?)

That should be ok. The patch was really merged by Mark and in upstream the commit id for the patch is:
```
commit 18ebffe4d043bf5f3a9b669d8d91f855bde8f6b7
Author: Iulian Olaru <iulianolaru249@yahoo.com>
Date:   Thu Sep 17 13:56:26 2020 +0300

    ASoC: SOF: imx: Add debug support for imx platforms
```

As we haven't done upstream sync after that was merged, we know it by a different commit-id still in sof-dev.

**Review by dbaluta — APPROVED** [review]

**Review by plbossart — APPROVED** [review]
I still don't get how that happned, we use

````
name: "BUILD GCC9 SOF Kernel arm64"
      script:
        - export ARCH=arm64 CROSS_COMPILE=/usr/bin/aarch64-linux-gnu- KCFLAGS="-
Wall -Werror"
````
so not sure what is needed to fail in such cases?

**Comment by kv2019i** [comment]
@plbossart wrote:
 > I still don't get how that happned, we use
> 
> ```
> name: "BUILD GCC9 SOF Kernel arm64"
>       script:
>         - export ARCH=arm64 CROSS_COMPILE=/usr/bin/aarch64-linux-gnu- KCFLAGS="-
> Wall -Werror"
> ```
> 
> so not sure what is needed to fail in such cases?

I checked modpost.c sources and it never returns an error in this case. There's an old patch from 2016 to have modpost optionally fail on warnings, but this was not accepted and no further attempts:
https://patchwork.kernel.org/patch/8343431/

**Comment by plbossart** [comment]
> @plbossart wrote:
> 
> > I still don't get how that happned, we use
> > ```
> > name: "BUILD GCC9 SOF Kernel arm64"
> >       script:
> >         - export ARCH=arm64 CROSS_COMPILE=/usr/bin/aarch64-linux-gnu- KCFLAGS="-
> > Wall -Werror"
> > ```
> > 
> > 
> > so not sure what is needed to fail in such cases?
> 
> I checked modpost.c sources and it never returns an error in this case. There's an old patch from 2016 to have modpost optionally fail on warnings, but this was not accepted and no further attempts:
> https://patchwork.kernel.org/patch/8343431/

Thanks for checking @kv2019i 
I guess this means we need to scan the logs for such problems. I am sure @marc-hb will have a script for that :-)

**Comment by marc-hb** [comment]
> Warning from Checkpatch (does CI do shallow pull and in reality that commit exists just fine?)

Yes checkpatch has serious issues with shallow clones (which do save a lot of CI time _when the PR has a recent base_), I already investigated and reported this in #2341 Of course checkpatch will never be able to deal with SHA1s that it cannot find, however it should at least sanity check the commit arguments and abort immediately, not try to find errors in only `$PERL_DEITY` knows what.

**Comment by marc-hb** [comment]
>  I am sure @marc-hb will have a script for that :-)

I take this as a compliment (thanks) but to be honest I hate screen-scraping logs for errors and warnings (who doesn't?) cause it's super brittle and time-consuming. A large part of my recent (scripting!) efforts in sof-test has been to rely on this less and on proper exit statuses more. I also made significant efforts to reduce the noise in the build logs (for instance [this script](https://github.com/thesofproject/sof/pull/3420) now has 600 lines out of output, down from 1200) with the assumption that developers sometimes look at build logs to spot warnings[*] because neither github nor Travis support warnings so looking at logs is the only way to spot warnings because you cannot turn every single warning everywhere into an error  otherwise you might as well kill the mere concept of warning.

In addition, my TODO list also has many things that unsurprisingly seem higher priority than this.

tl;dr: I recommend reviving https://patchwork.kernel.org/patch/8343431/ "modpost: allow modpost to fail on warnings"

This being said, I'm OK to help review a screen-scraping script if someone can create a reasonably short and not too convoluted one.

[*] I mean locally on their system. Incremental build logs are especially short and easy to look at.
